akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonProxy.scala

```diff
@@ -63,6 +63,7 @@ object ClusterSingletonProxySettings {
 /**
  * @param singletonName The actor name of the singleton actor that is started by the [[ClusterSingletonManager]].
  * @param role The role of the cluster nodes where the singleton can be deployed. If None, then any node will do.
+ * @param team The team of the cluster nodes where the singleton is running. If None then the same team as current node.
```

Comment:
Yes, makes sense in the context of Akka clustering. I'd rather say maybe not "catastrophic" but infrequent and perhaps possible to handle by other means (e.g. kafka topic) as my use case now involves handling this exact sort of failure.